crates/trie/trie/src/node_iter.rs

```diff
@@ -126,6 +131,27 @@ where
     ///
     /// If `metrics` feature is enabled, also updates the metrics.
     fn seek_hashed_entry(&mut self, key: B256) -> Result<Option<(B256, H::Value)>, DatabaseError> {
+        if let Some((last_key, last_value)) = self.last_next_result {
+            if last_key == key {
+                trace!(target: "trie::node_iter", seek_key = ?key, "reusing result from last next() call instead of seeking");
+                self.last_next_result = None; // Consume the cached value
+
+                let result = Some((last_key, last_value));
+                self.last_seeked_hashed_entry = Some(SeekedHashedEntry { seeked_key: key, result });
+
+                #[cfg(feature = "metrics")]
+                {
+                    self.metrics.inc_leaf_nodes_seeked();
+                    if Some(key) == self.previously_advanced_to_key {
+                        self.metrics.inc_leaf_nodes_same_seeked_as_advanced();
+                    }
+                }
```

Comment:
we didn't do an actual seek here, so this should be removed, as this metric is used for the actual number of database seeks that we made
```suggestion
```